### ISSUE TITLE
Allow configuration of tklbam-backup default cache location

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#179]
Following a bit of investigation to answer `Victor Vasev`'s question on the [TKL forums](http://www.turnkeylinux.org/forum/support/20140328/change-default-cache-location-tklbam-backups) I discovered that by default tklbam-backup stores its cache in /TKLBAM

I think that ideally the backup cache location should be configurable via /etc/tklbam/conf
##

**Comment by Vanessa45** [comment]
I use CloudBacko (http://free.cloudbacko.com?r=1d) as an automatic backup solution, then I do some manual backup once a month on a local hard drive. But I'd like to skip that manual thing and get instead a second automatic solution.

**Comment by JedMeister** [comment]
@Vanessa45 probably best to post on [our forums](http://www.turnkeylinux.org/forum) rather than here. We use this as our bug and feature request tracker and your post here doesn't really fit with this feature request anyway...

But wehile we're here, I'm not sure what OS you are using but if it's Debian then you can probably get [TKLBAM](http://www.turnkeylinux.org/docs/tklbam) to work with it - it'll take a bit of patience and tinkering on your end though...

**Comment by lirazsiri** [comment]
Yes we could do that. FWIW /TKLBAM doesn't hold a cache, it holds metadata about the backup (e.g., which files have changed, list of files to backup, list of packages, users and groups). These don't take up a lot of space but /TKLBAM also stores text dumps of the database. Those could potentially take up a lot of space - maybe more than you have on the root partition, in which case TKLBAM would get in trouble and fail.

While TKLBAM relies on /TKLBAM being in that particular location, we could get around this by adding support for mount binding /TKLBAM to any part of the filesystem that has more space.

Another thing to look into for the next maintenance round.
